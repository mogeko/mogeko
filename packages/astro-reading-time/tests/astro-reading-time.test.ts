import { describe, expect, it, vi } from "vitest";
import readingTime from "@/index";

const updateConfig = vi.fn(<T>(config: T): T => config);

describe("readingTime", () => {
  it("update the config", () => {
    const config = readingTime();

    config.hooks["astro:config:setup"]?.({ updateConfig } as any);

    expect(updateConfig).toHaveBeenCalled();
    expect(config.name).toBe("astro-reading-time");
    expect(config).toMatchSnapshot();
  });
});
