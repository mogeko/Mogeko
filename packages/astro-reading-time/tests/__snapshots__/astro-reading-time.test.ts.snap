// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`readingTime > update the config 1`] = `
{
  "hooks": {
    "astro:config:setup": [Function],
  },
  "name": "astro-reading-time",
}
`;
